src/main/java/seedu/eduke8/Parser.java

```diff
@@ -1,13 +1,27 @@
 package seedu.eduke8;
 
 
-import seedu.eduke8.quiz.QuizInterface;
+import seedu.eduke8.option.OptionInterface;
+import seedu.eduke8.question.QuestionInterface;
+import seedu.eduke8.quiz.Quiz;
+import seedu.eduke8.topic.TopicListInterface;
 import seedu.eduke8.ui.UiInterface;
 
+import java.util.ArrayList;
+
 public class Parser implements ParserInterface {
 
-    public void parseCommand(String userInput) {
+    public Parser() {
+    }
+
+    public void parseCommand(TopicListInterface topicList, String userInput) {
+        int numOfQuestions = 0;
+        String topicName = "";
         String[] commandArr = userInput.trim().split(" ", 0);
+        if (commandArr.length == 3 && commandArr[0].equals("quiz")) {
+            numOfQuestions = Integer.parseInt(commandArr[2].substring(2));
+            topicName = commandArr[1].substring(2);
+        }
```

Comment:
Great implementation for v1.0! Do take note that we have to handle the case where the input provided by the user is unfavorable though, such as if the input length is not 3.

Reply:
Alright, noted! Will look into how to make the implementation better in future!